src/Universe.cpp

```diff
@@ -1187,6 +1216,36 @@ std::map<int, Universe*> Lattice::getUniqueUniverses() {
 }
 
 
+/**
+ * @brief Get the maximum equivalent radius of each unique universes. Equivalent
+ *         radius are computed as the diagonal length to the cell boundary.
+ * @param unique_universes The unique universes of this Lattice
+ * @return a map of unique radius keyed by the universe ID.
+ */
+std::map<int, double> Lattice::getUniqueRadius
+                    (std::map<int, Universe*>  unique_universes) {
+
+  std::map<int, double> unique_radius;
+  std::map<int, Universe*>::iterator iter;
+  Universe* universe;
+  
+  for (iter = unique_universes.begin(); iter != unique_universes.end(); ++iter)
+    unique_radius[iter->first] = 0.;
+  
+  for (int k = _universes.size()-1; k > -1; k--) {
+    for (int j = _universes.at(k).size()-1; j > -1;  j--) {
+      for (int i = 0; i < _universes.at(k).at(j).size(); i++) {
+        universe = _universes.at(k).at(j).at(i).second;
+        unique_radius[universe->getId()] = 
+          std::max(unique_radius[universe->getId()],
+          sqrt(_widths_x[i]*_widths_x[i]/4.0 + _widths_y[j]*_widths_y[j]/4.0));
+      }
+    }
+  }
+  return unique_radius;
```

Comment:
So we have a different radius for every universe. I thought we were going to use the same radius everywhere

Reply:
For each kind of universe, it has its own equivalent radius